src/drivers/stm32/drv_io_timer.c

```diff
@@ -376,14 +380,39 @@ static int timer_set_rate(unsigned timer, unsigned rate)
 {
 
 	/* configure the timer to update at the desired rate */
-	rARR(timer) = 1000000 / rate;
+	rARR(timer) = timer_freq[timer] * 1000000 / rate;
 
 	/* generate an update event; reloads the counter and all registers */
 	rEGR(timer) = GTIM_EGR_UG;
 
 	return 0;
 }
 
+#define FAKE_ONESHOT
+void io_timer_set_oneshot_mode(unsigned timer)
+{
+	timer_freq[timer] = 8;
+
+#ifdef FAKE_ONESHOT
+	// to eliminate jitter with max additional latency of 500usec, set PWM rate to 2KHz.
+	// Note that the FMU mixer jitter is on the order of 2msec.
```

Comment:
The jitter is by design (trading execution speed for RAM). However, if we would run FMU again in its own thread we could get it down to ~20-40 us. I'm wondering if we should add the option for both (would be quite doable) and run FMU as thread in racers and in the work queue on less latency sensitive systems.